lesson01.md

```diff
@@ -12,5 +21,587 @@ Lesson 1, Tuesday, 2021-xx-xx
+
+### Pillars of Web Development
+
+| HTML | CSS | JavaScript |
+| ---- | --- | ---------- |
+| ![HTML5](images/HTML5_Logo.svg) <!-- .element height="64px" width="64px" --> | ![CSS3](images/css3.svg) <!-- .element height="64px" width="64px" --> | ![JS](images/javascript-logo.svg) <!-- .element height="64px" width="64px" --> |
+| *Content* |  *Presentation* | *Dynamic Effects* |
+| Nouns | Adjectives | Verbs |
+
+```html
+<p>Hey</p>  <!-- HTML: Adds a paragraph -->
+```
+
+```css
+p {color: red;}  /* CSS: Makes the paragraph red */
+```
+
+```js
+p.remove(); // JavaScript: removes the paragraph
+```
+
+---
+
+### JavaScript - What is it?
+
+* JS is a *programming language*
+* JS allows you to implement complex features on web pages
+* JS is the third layer of the cake of web technologies
+
+---
+
+* Interactivity?
+* Dynamic effects?
+* 2D/3D graphics?
+* Timed events?
+
+If you see any of this in a web page, it's (probably) JavaScript!
+
+---
+
+### JavaScript - "Full Stack"
+
+* For servers, back-end: [Node.js](https://nodejs.org/en/)
+* For mobile apps: [React Native](https://facebook.github.io/react-native/), [Cordova](https://cordova.apache.org/)
+* For desktop apps: [PWAs](https://developers.google.com/web/progressive-web-apps), [Elektron](https://www.electronjs.org/)
+
+
+---
+
+### Tools
+
+We'll be using these tools during our course:
+
+* [Chrome Browser](https://www.google.com/chrome/)
+* [Visual Studio Code](https://code.visualstudio.com/)
+* [MDN Web Documentation](https://developer.mozilla.org/en-US/)
+
+---
+
+### Tools: Chrome Browser
+
+* https://www.google.com/chrome/
+* Most common browser
+* Most up to date
+* Excellent Development Tools (F12)
+
+---
+
+### Tools: Visual Studio Code
+
+* https://code.visualstudio.com/
+* Open source IDE (**I**ntegrated **D**evelopment **E**nvironment)
+* Great tools
+* Good defaults
+* Fast
+
+---
+
+### Tools: MDN Web Documentation
+
+* https://developer.mozilla.org/en-US/
+* Up to date reference for JS
+* Good guides
+* Free
+
+The way I use it:
+<!-- .element: class="fragment" data-fragment-index="1" -->
+- open my favourite search engine
+<!-- .element: class="fragment" data-fragment-index="1" -->
+- type <!-- .element: class="fragment" data-fragment-index="1" --> `mdn [topic I'm looking for]`
+<!-- .element: class="fragment" data-fragment-index="1" -->
+- for example: <!-- .element: class="fragment" data-fragment-index="1" --> `mdn javascript variables`
+<!-- .element: class="fragment" data-fragment-index="1" -->
+
+---
+
+### Additional Material
+
+* [javascript.info](https://javascript.info/) - tutorials with nice (often visual) explanations  
+* [freecodecamp.org](https://www.freecodecamp.org/) - full free courses on many topics
+* [udemy.com](https://www.udemy.com): Introduction to JS (course)
+* [codeacademy.com](https://www.codecademy.com): Introduction to JS (another course)
+* [edabit.com](https://edabit.com): Learn JavaScript with interactive challenges and external resources
+
```

Comment:
I recently discovered a new JS course, you have to pay 40 EUR for the complete course, but it is kick-ass compared to freecodecamp and udemy and Codecademy. I wonder shall we add that to the list?
https://learnjavascript.online

Reply:
I can add it and specify that it's not for free